extension.js

```diff
@@ -125,24 +126,29 @@ var ContainerSubMenuMenuItem = GObject.registerClass(
     class extends PopupMenu.PopupSubMenuMenuItem {
         _init(container) {
             super._init(container.name);
-            this.menu.addMenuItem(new PopupMenuItem("Status", container.status));
-            this.menu.addMenuItem(new PopupMenuItem("Id", container.id));
-            this.menu.addMenuItem(new PopupMenuItem("Image", container.image));
-            this.menu.addMenuItem(new PopupMenuItem("Command", container.command));
-            this.menu.addMenuItem(new PopupMenuItem("Created", container.createdAt));
-            this.menu.addMenuItem(new PopupMenuItem("Ports", container.ports));
-            const ipAddrMenuItem = new PopupMenuItem("IP Address", "");
-            this.menu.addMenuItem(ipAddrMenuItem);
-            this.inspected = false;
-
-            // add more stats and info - inspect - SLOW
-            this.connect("button_press_event", () => {
-                if (!this.inspected) {
-                    container.inspect();
-                    this.inspected = true;
-                    ipAddrMenuItem.label.set_text(`${ipAddrMenuItem.label.text} ${container.ipAddress}`);
-                }
-            });
+            const actions = new PopupMenu.PopupBaseMenuItem({reactive: false, can_focus: false});
+            const details = new PopupMenu.PopupMenuSection();
+            this.menu.addMenuItem(actions);
+            this.menu.addMenuItem(new PopupMenu.PopupSeparatorMenuItem());
+            this.menu.addMenuItem(details);
+
+            const startBtn = addButton(actions, () => container.start(), "media-playback-start-symbolic");
+            const stopBtn = addButton(actions, () => container.stop(), "media-playback-stop-symbolic");
+            const restartBtn = addButton(actions, () => container.restart(), "system-reboot-symbolic");
+            const deleteBtn = addButton(actions, () => container.rm(), "user-trash-symbolic.symbolic");
```

Comment:
Two things worth considering:
 - color the icon red to mark the action as destructive (via a style class on the button)
 - add some form of confirmation

Reply:
I'm using https://gitlab.gnome.org/GNOME/gnome-shell/-/blob/main/js/ui/modalDialog.js. Is this the right class to use?

Reply:
It's the right class for a system dialog. That's the easiest option for confirmation, with the caveat that those dialogs are a bit intrusive. It's probably OK, assuming that deleting containers isn't the most frequent action (but maybe that's just me?).

An alternative could be to hide the container and show an "undo" notification, then delete the container for real when the notification closes.

Reply:
the 'undo' suggestion is nice. I'll need to make sure the dialog stays open and is synced while open - something I didn't implement so far. Any general guildelines about sync? maybe point me to some good example of a sync implementation.

Reply:
anyhow, I'll upload the updated PR with the dialog for initial review, I tried to make its implementation minimal. 
It can be midterm solution till I'll have a proper undo experiene

Reply:
>  I'll need to make sure the dialog stays open and is synced while open

Mmh, not sure I understand that, sorry.

All our dialogs are modal, so the user is forced to take a decision (unless there's a timeout like in the log-out/power-off dialog, but I don't think that pattern applies here).

The 'undo' pattern would use a notification instead, which allows users to go on with their job in the likely case that they don't want to cancel the operation.

That said:

> I'll upload the updated PR with the dialog for initial review

That's entirely reasonably at least initially, if not long-term. It depends a bit on whether you expect containers to be more of a throw-away thing (i.e. created and removed frequently, contents easily reproduced in case of accidental removal etc.) or more curated.

Reply:
I expect that quite a lot of the use cases would be that containers are long running, persistent daily tools and not throw-away experiments. Think of developers running postgres, nginx containers, or any other combination.  Toolbox will also be one such container that every Silverblue user would probably see (multiple toolbox probably). 
Confirmation for deletion is a much needed behaviour.

I want to make sure I understand your other suggestion - reading again what you wrote I understand that the 'undo' behaviour is something like: press the trash button, menu is closed, get a main notification (Main.notify?) that the container is removed unless you click the message. On message close invoke `container.rm()` - is this the direction?